src/py42/clients/archive.py

```diff
@@ -184,16 +193,19 @@ def stream_to_device(
             overwrite_existing_files,
         )
 
-    def _select_backup_set_id(self, device_guid, destination_guid):
+    def _select_backup_set_id(self, device_guid, destination_guid, backup_set_id):
         backup_sets = self.get_backup_sets(device_guid, destination_guid)["backupSets"]
-        if not backup_sets:
-            return None
-        for backup_set in backup_sets:
-            backup_set_id = backup_set["backupSetId"]
-            if str(backup_set_id) == _DEFAULT_BACKUP_SET_ID:
-                return backup_set_id
-
-        return backup_sets[0]["backupSetId"]
+        backup_set_ids = [bs["backupSetId"] for bs in backup_sets]
+        if backup_set_id:
+            if backup_set_id not in backup_set_ids:
+                raise Py42Error(f"{backup_set_id=} not found in device backup sets: {backup_sets}")
```

Comment:
`{backup_set_id=}` oops

Reply:
That's actually a feature of f-strings:

```
Python 3.8.10 (default, Jun  9 2021, 16:13:34) 
[Clang 12.0.5 (clang-1205.0.22.9)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> x = 1
>>> print(f"{x=}")
x=1
```

But in looking it up again I did realize it was introduced in python3.8 so we can't use it here anyway.

Reply:
Whoops yeah should’ve mentioned it looked like it was only failing on earlier python versions 👍